src/main/scala/com/ambiata/disorder/EncodingString.scala

```diff
@@ -0,0 +1,23 @@
+package com.ambiata.disorder
+
+import org.scalacheck._, Arbitrary._, Gen._
+import scala.io.Codec
+import scalaz._, Scalaz._
+import Arbitraries._
+import GenPlus._
+
+/**
+  Produce a string which is valid for the associated codec
+  */
+case class EncodingString(value: String, codec: Codec)
+
+object EncodingString {
+  implicit def EncodingStringArbitrary: Arbitrary[EncodingString] =
+    Arbitrary(checkGenTuple[EncodingString, S, Codec](
+        (s, c) => EncodingString(s.value, c)
+      , "EncodingString"
+      , es => validForCodec(es.value, es.codec)))
+
+  def validForCodec(s: String, c: Codec): Boolean =
```

Comment:
As mentioned IRL, the alternative is we define the supported/known character ranges for UTF8 and ISO8859. Not sure if that's better or worse though?